packages/builders/src/interactions/slashCommands/mixins/NameAndDescription.ts

```diff
@@ -31,4 +35,86 @@ export class SharedNameAndDescription {
 
 		return this;
 	}
+
+	/**
+	 * Sets a name localization
+	 *
+	 * @param locale The locale to set a description for
+	 * @param localizedName The localized description for the given locale
+	 */
+	public setNameLocalization(locale: LocaleString, localizedName: string | null) {
+		if (!this.name_localizations) {
+			Reflect.set(this, 'name_localizations', {});
+		}
+
+		validateLocale(locale);
+
+		if (localizedName === null) {
+			this.name_localizations![locale] = null;
+			return this;
+		}
+
+		validateName(localizedName);
+
+		this.name_localizations![locale] = localizedName;
+		return this;
+	}
+
+	/**
+	 * Sets the name localizations
+	 *
+	 * @param localizedNames The dictionary of localized descriptions to set
+	 */
+	public setNameLocalizations(localizedNames: Map<LocaleString, string> | LocalizationMap | null) {
```

Comment:
@discordjs/the-big-4 should we support maps too? I remember when I did this and then we decided against it for choices

Reply:
I don't see why not, we can turn Map into plain objects with `Object.fromEntries(map)`

Reply:
I for one argue we should support just what Discord expects, keep everything clean

Reply:
I agree with Vlad here, I'll probably just make it accept objects only.

Reply:
Sure thing 👍🏼